analytics/msf4j-analytics/src/main/java/org/wso2/msf4j/analytics/httpmonitoring/config/HTTPMonitoringConfigBuilder.java

```
@@ -15,29 +15,61 @@
+                throw new RuntimeException(
+                        "Failed to populate HTTPMonitoringConfig Configuration. Config Provider is Null.");
+            }
+            //Standalone mode
+            String deploymentYamlPath = System.getProperty(DEPLOYMENT_YAML_SYS_PROPERTY);
+            if (deploymentYamlPath == null || deploymentYamlPath.isEmpty()) {
+                logger.info("System property '" + DEPLOYMENT_YAML_SYS_PROPERTY +
+                            "' is not set. Default deployment.yaml file will be used.");
+                deploymentYamlPath = HTTPMonitoringConfig.class.getResource("/deployment.yaml").getPath();
+            } else if (!Files.exists(Paths.get(deploymentYamlPath))) {
+                String msg = "Couldn't find " + deploymentYamlPath;
+                throw new RuntimeException(msg);
+            }
+
             try {
-                Yaml yaml = new Yaml();
-                return yaml.loadAs(metricsConfigFileContent.get(), HTTPMonitoringConfig.class);
-            } catch (RuntimeException e) {
-                throw new RuntimeException("Failed to populate HTTP Monitoring Configuration", e);
+                configProvider = ConfigProviderFactory.getConfigProvider(Paths.get(deploymentYamlPath), null);
+                DataHolder.getInstance().setConfigProvider(configProvider);
+            } catch (ConfigurationException e) {
+                throw new RuntimeException("Error loading deployment.yaml Configuration", e);
             }
-        } else {
-            return new HTTPMonitoringConfig();
         }
+
+        try {
+            configurationObject =
+                    DataHolder.getInstance().getConfigProvider().getConfigurationObject(HTTPMonitoringConfig.class);
+        } catch (ConfigurationException e) {
+            logger.error("Error loading configurations from deployment.yaml", e);
+            configurationObject = new HTTPMonitoringConfig();
```

New instantiation can again cause an exception. Better to throw a runtime exception as in the previous case.